Разметка главной страницы

Redisco1 opened the pull request:
---
:mortar_board: [Разметка главной страницы](https://up.htmlacademy.ru/htmlcss/33/user/1554631/tasks/3)

:boom: https://htmlacademy-htmlcss.github.io/1554631-sedona-33/2/

Redisco1 committed:
Создал 2-ю страницу сайта "Каталог"

Redisco1 committed:
Сверстал header

Redisco1 committed:
Исправил подпись иконки в Header

Redisco1 committed:
Разметил главную страницу

Точно не понял к какому разделу отнести картинки с приветствием, которые идут после Header, решил создать отдельный раздел <section class= "welcome".

Redisco1 commented on index.html:
```diff
@@ -37,21 +43,133 @@
   <main class="main-index">
-    <section class="">
-
+    <section class="welcome">
```
Не понял в какой раздел отнести Картинки с приветствием, которые идут сразу после Хедера, думал к разделу "Преимущества", но решил создать отдельный section class=welcome

alicepepel replied:
Я бы тоже создала отдельный раздел. Жестких правил нет на этот счет, ты делишь макет по смысловым блокам... и кто-то увидит здесь один раздел, а кто-то два ) Мне кажется, вэлком - очень подходящее название для этой секции ;)

Redisco1 commented on index.html:
```diff
@@ -37,21 +43,133 @@
+        <p class="advantages-start-subtitle">Рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</p>
+      </div>
+      <ol class="advantages-list">
+        <li class="advantages-conteiner-item">
```
Тут тоже не понял, как лучше сделать, тк 1 чистым списком не получалось все преимущества из-за картинки, поэтому было 2 выбора, которые пришли мне в голову: 1) список в списке и 2) сделать элементы списка с фотографией отдельным блоком через div (я выбрал 2-й вариант ). Другие варианты мой мозг не придумал, еще недавно увидел, что можно через

сделать, но это было уже после того, как запушил и мне интересен мой вариант.

alicepepel replied:
Да, у тебя здорово размечен этот блок. В списке можно использовать div-ы, и это не какая-то плохая практика )

Единственное, я бы использовала здесь ul. Нумерованный список ol используется, когда важен порядок элементов, и тут он вроде есть... Но вклиниваются жилье-еда-сувениры, они ведь не пронумерованы. И ломают логику ol. Ну это мы еще на созвоне обсудим, блок непростой )

alicepepel commented on index.html:
```diff
@@ -3,14 +3,15 @@
 
 <body>
   <header claass="page-header">
     <nav class="novigation">
```
Очепятка, видимо - nAvigation ) Замени во всем документе novigation-ы.

alicepepel commented on index.html:
```diff
@@ -20,15 +21,20 @@
+          <a class="novigation-link" href="#">
+            <span class="visually-hidden">Избранное</span>
+          </a>
+          <sup class="novigation-counter">12</sup>
```
О, здорово, что ты нашел такой редко используемый тег ) И что вообще заметил эту цифру. Действительно, ее стоило внести в разметку )

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
-    <section class="">
-
+    <section class="welcome">
+      <h1 class="visually-hidden">Welcome Sedona</h1>
```
h1 есть, это хорошо. Но поскольку сайт весь на русском языке, логичнее предположить, что и главный заголовок должен быть на русском ) Ведь его будут читать скринридеры и поисковые системы соответствующей страны. Можно что-то типа "сайт о городе Седона", ну или просто Седона )

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+      <div class="welcome-block-img">
+        <img class="welcome-img" src="" alt="Добро пожаловать в  Седону" />
+        <img class="welcome-divider" src="" />
+      </div>
```
Когда созвонимся, спрошу, почему тут картинки лежат в div-е ) Никакой критичной ошибки, просто можно было обойтись и без этой обертки. Интересно, почему ты решил ее использовать.

Upd: у меня есть предположение. Чтобы создать тень по бокам макета? Если да, это обсудим тоже )

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+      <div class="advantages-start-description">
+        <h3 class="advantages-start-title">Седона — небольшой городок в Аризоне, заслуживающий большего!</h3>
+        <p class="advantages-start-subtitle">Рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</p>
+      </div>
```
В принципе, тут тоже не обязательно оборачивать h3 и p в общий div. Это просто два текста, которые идут друг за другом, никаких сложностей в дизайне, визуально они не сгруппированы.

А вот дальше идет группировка, и это бросается в глаза - все преимущества разбиты на квадраты, которые составляют большую мозаику. Они не просто расположены друг под другом, спокойно ) Они как-то взаимодействуют, прижимают друг друга, по вертикали и горизонтали. А h3 и p - неа.

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+          гостиниц в Седоне</p>
+        <div class="hotel-search-button">
+          <a class="hotel-search-button" href="#">ПОИСК ГОСТИНИЦЫ В седоне</a>
+        </div>
```
А почему здесь обернул кнопку в отдельный div? )

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+    <section class="newsletter">
+      <div class="newsletter-container">
+        <h2 class="visually-hidden">Рассылка</h2>
+        <h3 class="newsletter-title">Подпишитесь на рассылку</h3>
```
Вообще, не обязательно во всех блоках делать одинаковую структуру заголовков. В предыдущих секциях заголовки были какие-то непонятные, не говорящие о содержании секции. Поэтому мы использовали дополнительный h2 visually-hidden, чтобы там написать, о чем этот раздел.

Здесь же в макете - нормальный, содержательный заголовок "подпишитесь на рассылку". Можно сразу его сделать h2, видимым. Не нужно дублировать h2 и h3. Надеюсь, более или менее понятно )

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+      <h2 class="visually-hidden">Номер телефона</h2>
+      <div class="footer-number-block">
+        <a class="footer-number-tel" href="tel:+78128121212">+7 (812) 812-12-12</a>
+      </div>
```
Не обязательно оборачивать ссылку еще и div-ом ) Она просто лежит в section, одна-единственная, ее не нужно ни от чего обособлять.

alicepepel commented on index.html:
```diff
@@ -37,21 +43,133 @@
+        <a target="target" class="footer-logo-link" href="https://htmlacademy.ru/">
+          <img class="footer-logo-img" src="#" alt="Логотип HTMLAcademy" />
+        </a>
+      </div>
```
Та же самая история с div - это просто одиноко лежащая ссылка. Мы стараемся не делать лишних оберток, чтобы было проще разбираться в коде. Поэтому, если див можно убрать - убираем )

Redisco1 committed:
Внес коррективы

Исправил все недочеты отмеченные наставником, изменил разметку сектора advantages

Redisco1 committed:
Merge branch 'htmlacademy-htmlcss:master' into master
